### ISSUE TITLE
Any Plan setting box-in mask-out training first?

### ISSUE THREAD

**Issue by awerty-noob** [Joey-S-Liu/MedSAM3#8]
It seems that the box-based input, as demonstrated in MedSAM-2, plays a critical role in improving segmentation quality, especially when dealing with intricate anatomical structures. Based on the results in MedSAM-3, I observed that the performance with text-only prompts (MedSAM-3 T) was limited across various datasets, whereas combining the textual prompts with bounding box input (MedSAM-3 T+I) led to significant performance gains.

Do you foresee incorporating box-based training as a first step before fine-tuning MedSAM-3 with PCS-style training? Given that this might strengthen the alignment between the model's segmentation outputs and anatomical regions, I believe it could improve the performance of PCS tasks as well. This could help tackle challenges like fine-grained medical terminology alignment and anatomical misalignment, which MedSAM-3 sometimes struggles with.

Any insights or plans regarding this would be greatly appreciated!

Thanks.

**Comment by Joey-S-Liu** [comment]
> It seems that the box-based input, as demonstrated in MedSAM-2, plays a critical role in improving segmentation quality, especially when dealing with intricate anatomical structures. Based on the results in MedSAM-3, I observed that the performance with text-only prompts (MedSAM-3 T) was limited across various datasets, whereas combining the textual prompts with bounding box input (MedSAM-3 T+I) led to significant performance gains.
> 
> Do you foresee incorporating box-based training as a first step before fine-tuning MedSAM-3 with PCS-style training? Given that this might strengthen the alignment between the model's segmentation outputs and anatomical regions, I believe it could improve the performance of PCS tasks as well. This could help tackle challenges like fine-grained medical terminology alignment and anatomical misalignment, which MedSAM-3 sometimes struggles with.
> 
> Any insights or plans regarding this would be greatly appreciated!
> 
> Thanks.

Thanks for your interest. Please try our MedSAM3_v1 weight (training with pure text on 330 tasks), which we released a few days ago. We will update the arXiv paper after the MedSAM3_v2 version.

**Comment by awerty-noob** [comment]
Well, I tried your v1 weight on CXR, and do segment like `xrv.baseline_models.chestx_det.PSPNet`, but got the poor result: It basically stick with the lung and barely recognize other structure, not to mention any abnormals. I ran the validate script to see if there was any config oops but no error. Is that a expected behavior?

**Comment by Joey-S-Liu** [comment]
> Well, I tried your v1 weight on CXR, and do segment like `xrv.baseline_models.chestx_det.PSPNet`, but got the poor result: It basically stick with the lung and barely recognize other structure, not to mention any abnormals. I ran the validate script to see if there was any config oops but no error. Is that a expected behavior?

What is your text prompt? The text prompt may be out of v1's 330 settings.

**Comment by awerty-noob** [comment]
<img width="1135" height="1183" alt="Image" src="https://github.com/user-attachments/assets/6331bbd3-ac61-42bd-80b6-5736f5be59e5" />

**Comment by Joey-S-Liu** [comment]
> <img alt="Image" width="1135" height="1183" src="https://private-user-images.githubusercontent.com/36948728/546709707-6331bbd3-ac61-42bd-80b6-5736f5be59e5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzA1MzUxNDEsIm5iZiI6MTc3MDUzNDg0MSwicGF0aCI6Ii8zNjk0ODcyOC81NDY3MDk3MDctNjMzMWJiZDMtYWM2MS00MmJkLTgwYjYtNTczNmY1YmU1OWU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjA4VDA3MTQwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcyOWM4MmYyM2MyY2E2YWE2NDdhZGNhMTI4NTU1M2JlOWNlYjY1YWIyMzRkYzhiNWFlODU5MGU1ZThkZTViZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.M_Hq1XspmIfX-a0-URNysE3SsEDRE0q75Zcsh6GGT6s">

We didn't cover this word in chest X-ray.

**Comment by awerty-noob** [comment]
Okay, looks like a new bunch of training data can be generated :)
Given the sheer volume of vocabulary in medical structures, 330 is indeed a very small number.

**Comment by IrohXu** [comment]
> Okay, looks like a new bunch of training data can be generated :) Given the sheer volume of vocabulary in medical structures, 330 is indeed a very small number.

You are right. More data is continuous training in our cluster. We will release version 2.0 soon.

**Comment by Yumonor** [comment]
You mentioned using 330 Unique Medical Text IDs to train the model. Could you please provide those medical text IDs?
Thank you for providing your excellent work.

**Comment by enescicek** [comment]
@Joey-S-Liu 
@IrohXu 

Hello, first of all, congratulations on this excellent work. I am using MedSAM3 to segment teeth in radiographic dental images, but the results are very unsuccessful. I used ‘teeth’ as the prompt. What should I write in the prompt to achieve better results? Alternatively, is there a way to perform segmentation with MedSAM3 by drawing a bounding box?

**Comment by enescicek** [comment]
@awerty-noob 

Hello, could you please explain how you used the bounding box + text prompt together in your MedSAM3 study? When I used only a text prompt, I obtained very unsuccessful results. I would like to try experimenting with bounding box input as well.

**Comment by Mo-Junyang** [comment]
@Joey-S-Liu hello, Are you going to train a version with bounding box? I try to segment a ultrasound image with bounding box input, but the results are very bad.

**Comment by Lee728243228** [comment]
@awerty-noob @enescicek @Joey-S-Liu Hello, I have the same question, Could you explain how can I  used the bounding box + text prompt together in MedSAM3 study?
